### PR TITLE
NOISSUE - update mqtt prov tool and some refactor

### PR DESCRIPTION
**Pull request by mteodor**
### What does this do
update mqtt provision tool to connect each thing with each channel created. refactor some structures

### CODE REVIEW

**Comment by drasko** [comment]
@mteodor if you plan to keep `client` receiver, please implement `client.New()` client factory.

**Comment by drasko** [comment]
@anovakovic01 @dusanb94 is keeping `client` receiver a good idea, having in mind that everything moved to the base `bench` package now?

File: tools/mqtt-bench/bench.go [hunk 1]
```diff
@@ -148,12 +152,14 @@ func Benchmark(cfg Config) {
 
 	wg.Wait()
 
+	start := time.Now()
 	// Publishers
 	for i := 0; i < cfg.Test.Pubs; i++ {
-		mfConn := mf.Conns[i%n]
+		mfConn := mf.Channels[i%n]
```
Comment (dborovcanin) [inline tools/mqtt-bench/bench.go]: Using `mfChann` instead of `mfConn` makes more sense.

File: tools/provision/provision.go [hunk 2]
```diff
@@ -205,6 +209,23 @@ func Provision(conf Config) {
...
+	fmt.Println("# each channel is connected to each thing from things list")
+	for i := 0; i < conf.Num; i++ {
+		for j := 0; j < conf.Num; j++ {
+			wg.Add(1)
+			go func(wg *sync.WaitGroup, i, j int) {
+				defer wg.Done()
+				s.ConnectThing(things[j].ID, *channels[i], token)
+			}(&wg, i, j)
+		}
+		fmt.Println("[[channels]]")
```
Comment (dborovcanin) [inline tools/provision/provision.go]: A single `fmt.Printf(...)` instead of multiple `fmt.Println(...)`?

**Review by dborovcanin — DISMISSED** [review]

File: tools/mqtt-bench/bench.go (hunk 1 above)
Comment (drasko) [reply tools/mqtt-bench/bench.go]: Agreed

File: tools/provision/provision.go [hunk 3]
```diff
@@ -116,13 +117,20 @@ func Provision(conf Config) {
 	}
 
 	//  Create things and channels
+	things := make([]*sdk.Thing, conf.Num)
+	channels := make([]*string, conf.Num)
+
+	fmt.Println("# list of things that can be connected to MQTT broker")
```
Comment (drasko) [inline tools/provision/provision.go]: Comments start with capital letters

File: tools/provision/provision.go (hunk 2 above)
Comment (drasko) [reply tools/provision/provision.go]: Agreed

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by dborovcanin** [comment]
> @anovakovic01 @dusanb94 is keeping `client` receiver a good idea, having in mind that everything moved to the base `bench` package now? 

For me, it makes sense to keep the client, at least for now.

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**